### PR TITLE
Update zend-diactoros version requirement

### PR DESCRIPTION
**Pull request by krisolafson**
Needed when updating requirements for N3-3049 at https://github.com/acquia/network-n3/pull/1427

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "require": {
         "php": "~5.6 || ~7.1",
         "willdurand/negotiation": "~2.3.1",
-        "zendframework/zend-diactoros": "~1.1.0"
+        "zendframework/zend-diactoros": "^1.7"
```
Comment (itafroma) [inline composer.json]: It works with 1.1: let's just update this to `^1.1`

Comment (krisolafson) [reply composer.json]: N3 framework wants "zendframework/zend-diactoros": "^1.7"

Do we have to match that?

Comment (itafroma) [reply composer.json]: No: Composer will be able to resolve the dependency to `1.7.0` or higher since `^1.1` is equivalent to `>=1.1.0 && < 2.0.0`.

Comment (krisolafson) [reply composer.json]: ok will adjust!

**Pull request merged**